### ISSUE TITLE
RustDesk build panics at "could not find system library 'xi'"

### ISSUE THREAD

**Issue by Occhioverde** [NixOS/nixpkgs#197977]
### Steps To Reproduce
1. Add `pkgs.rustdesk` to your `home.packages` in Home Manager or in your `environment.systemPackages` in NixOS module
2. Run `home-manager switch` or `nixos-rebuild switch`
3. See the error in log

### Build log
```console
❯ home-manager switch
error: builder for '/nix/store/nkhzbg9m85vphpv8i075rxkhfv9d0m18-rustdesk-1.1.9.drv' failed with exit code 101;
       last 10 log lines:
       >   --- stderr
       >   thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: `"pkg-config" "--libs" "--cflags" "xi" "xi >= 1.7"` did not exit successfully: exit status: 1
       >   error: could not find system library 'xi' required by the 'x11' crate
       >
       >   --- stderr
       >   Package file '/nix/store/zxld20wh65ca5aazxv3hwzm3saqskr4a-libXi-1.8-dev/lib/pkgconfig/xi.pc' appears to be empty
       >   Package 'xi' has no Name: field
       >   ', /build/rustdesk-1.1.9-vendor.tar.gz/x11/build.rs:40:14
       >   note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
       > warning: build failed, waiting for other jobs to finish...
       For full logs, run 'nix log /nix/store/nkhzbg9m85vphpv8i075rxkhfv9d0m18-rustdesk-1.1.9.drv'.
error: 1 dependencies of derivation '/nix/store/flnx4xkr3fg3d2bhkx45ww407hki2fwc-home-manager-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/qxlzx5bb8zlqw99jaf18qbfzdm3ds5zf-man-paths.drv' failed to build
error: 1 dependencies of derivation '/nix/store/k6mnc9zsz04ni84mlbfqpnx5g17c7jw5-rustdesk-1.1.9-fish-completions.drv' failed to build
error: 1 dependencies of derivation '/nix/store/4iascx8n2d6bflym4866gg3jx1kq89jf-home-manager-generation.drv' failed to build
```

### Additional context
The Home Manager Nix Flake is locked on Nixpkgs Unstable rev. f994293d1eb8812f032e8919e10a594567cf6ef7

It may be worthwhile to point out that my system is currently running on a Wayland composer and that X11 is only present as XWayland.

### Notify maintainers

@Leixb

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
root@Ricky-First-Laptop /e/nixos# nix-info -m
 - system: `"x86_64-linux"`
 - host os: `Linux 5.19.12-zen1, NixOS, 22.11 (Raccoon), 22.11.20221013.ba187fb`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - channels(root): `"nixos-22.05"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by Leixb** [comment]
I cannot reproduce the error right now. Can you try this override and see if it fixes the issue:
```nix
(rustdesk.overrideAttrs (oldAttrs: {
  buildInputs = oldAttrs.buildInputs ++ [ pkgs.xorg.libXi ];
}))
```

**Comment by Occhioverde** [comment]
Thank you very much for the quick reply! I just opened my computer and tried to apply the override, but the result is the exact same.

> Keep in mind that `rustdesk` does not work in wayland as far as I know

Yes, RustDesk is unable to record the screen and to send keystrokes to most of the Wayland composers, but, since I plan to only connect to remote PCs, this limitation shouldn't be a problem.

**Comment by Leixb** [comment]
The issue is probably that the file at: `/nix/store/zxld20wh65ca5aazxv3hwzm3saqskr4a-libXi-1.8-dev/lib/pkgconfig/xi.pc` should not be empty, probably there is some error in the filesystem. Try `sudo nix store repair /nix/store/zxld20wh65ca5aazxv3hwzm3saqskr4a-libXi-1.8-dev` and on any other paths causing issues.

I would also check the system logs and the filesystem integrity to make sure the drive is not failing

**Comment by Occhioverde** [comment]
> Try `sudo nix store repair /nix/store/zxld20wh65ca5aazxv3hwzm3saqskr4a-libXi-1.8-dev` and on any other paths causing issues.

This seems to have solved the issue!

From what I understand, some paths in the Nix Store became corrupted as a result of an unexpected shutdown that occurred during a download, causing the problem I encountered.

In order to avoid issues with other derivations I ran `sudo nix-store --verify --check-contents --repair` to repair my entire store, so now everything should be good.

In the future I will check for corruptions before opening a ticket.

Thank you very much for your help!
